src/models/room-state.ts

```diff
@@ -607,13 +608,15 @@ export class RoomState extends EventEmitter {
     /**
      * Returns true if the given user ID has permission to send a normal
      * event of type `eventType` into this room.
-     * @param {string} eventType The type of event to test
+     * @param {string} eventType The type of event to test.
      * @param {string} userId The user ID of the user to test permission for
      * @return {boolean} true if the given user ID should be permitted to send
      *                        the given type of event into this room,
      *                        according to the room's state.
      */
-    public maySendEvent(eventType: EventType | string, userId: string): boolean {
+    public maySendEvent<S extends string, T extends string>(
+        eventType: EventType | string | UnstableValue<S, T>,
```

Comment:
These functions should not be taking `UnstableValue` directly. The caller will need to call the function for both types if it wants to.

This should also remove the iterator requirement on `UnstableValue` - something that shouldn't be present there.

Reply:
> These functions should not be taking `UnstableValue` directly. The caller will need to call the function for both types if it wants to.

Why? It feels to me like this is both a common operation when prototyping events and something error-prone (and slower) to reimplement manually.

> This should also remove the iterator requirement on UnstableValue - something that shouldn't be present there.

I don't understand that remark, either.

Reply:
The logic would end up duplicated anyway as the user may have permission to send one but not the other so the called needs to know which one in the case of knowing whether it may have to send the unstable one because thats the only one it has perms to send to, if that's even a thing it wishes to do.

Reply:
> 

How's that different from the code I wrote that we're discussing?

Reply:
Behaviourally it's fine for your PR, but it isn't necessarily safe for other implementations or features. It's very possible that a feature down the line needs explicitly one or the other, and this signature would introduce a potential false sense of security to a developer.

Reply:
Ok, I've changed the implementation.